### PR TITLE
Make StateMacheineStates variables Constants

### PR DESCRIPTION
**Pull request by AdamSpeight2008**

### CODE REVIEW

**Comment by AdamSpeight2008** [comment]
Alternatively we could make the Enum values and move them into the `StateMachineRewriter` class itself.
 
```vbnet
Namespace Microsoft.CodeAnalysis.VisualBasic
    Partial Friend MustInherit Class StateMachineRewriter(Of TProxy)

        Friend Enum StateMachineStates As Integer
            FinishedStateMachine = -2
            NotStartedStateMachine = -1
            FirstUnusedState = 0
        End Enum

    End Class

End Namespace
```

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb [hunk 1]
```diff
@@ -1,5 +1,5 @@
-﻿' Licensed to the .NET Foundation under one or more agreements.
-' The .NET Foundation licenses this file to you under the MIT license.
+' Licensed to the .NET Foundation under one or more agreements.	
+' The .NET Foundation licenses this file to you under the MIT license.
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: revert.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿' Licensed to the .NET Foundation under one or more agreements.
+' Licensed to the .NET Foundation under one or more agreements.
```
Comment (CyrusNajmabadi) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: please revert this.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb [hunk 3]
```diff
@@ -10,9 +10,9 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 Namespace Microsoft.CodeAnalysis.VisualBasic
 
     Friend Module StateMachineStates
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: >Friend Module StateMachineStates [](start = 4, length = 32)

I think that at this point we can make this an Enum. Is there a reason not to?

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb (hunk 3 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: @AlekseyTs I did try an implement and it worked, compiled and passed the unittest on my machine. The proposed implementation is in a reply above.

Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: All issue is if there are any external use _i can't observe_ that could be affected by the change of type / representation.

Comment (CyrusNajmabadi) [reply src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: It's an internal type in an assembly that isn't available for IVT.  So you should be fine to make this change.  thanks :)

Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: > Only issue is if there are any external use i can't observe that could be affected by the change of type / representation.

We generally think that we are free to change internal/friend declaration as we wish. 

---
In reply to: [400488739](https://github.com/dotnet/roslyn/pull/42870#discussion_r400488739) [](ancestors = 400488739)

**Comment by AdamSpeight2008** [comment]
Push in class and made an enum.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb [hunk 4]
```diff
@@ -8,11 +8,14 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
...
 
 Namespace Microsoft.CodeAnalysis.VisualBasic
+    Partial Friend MustInherit Class StateMachineRewriter(Of TProxy)
 
-    Friend Module StateMachineStates
-        Public FinishedStateMachine As Integer = -2
-        Public NotStartedStateMachine As Integer = -1
-        Public FirstUnusedState As Integer = 0
-    End Module
+        Friend Enum StateMachineStates As Integer
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: >Friend [](start = 8, length = 6)

Protected?

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

File: src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb (hunk 4 above)
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb]: >StateMachineStates [](start = 20, length = 18)

It feels wrong to use an enum for these values, as the values that state machine states take are not restricted to these three values. Typical state machines have states with the value 1, 2, 3, etc.  It is just that those values and the part of the state machine they refer to are computed on the fly.

**Pull request closed without merging**